src/smartpeak/source/core/SessionHandler.cpp

```diff
@@ -2007,121 +2060,122 @@ namespace SmartPeak
-              std::vector<float> y_fit_data;
-              result.quant_methods.push_back(quant_method);
-              if (calibration_curve_found)
-              {
-                for (const auto& ratio : stand_concs_map.at(quant_method.getComponentName()).first) {
-                  // TODO: encapsulate in its own method e.g. sequenceSegmentProcessor
-                  // TODO: check that the calibration actually found a best fit (and set to all 0 if not)
-                  // calculate the absolute concentration
-                  OpenMS::TransformationModel::DataPoints data;
-                  OpenMS::TransformationDescription tmd(data);
-                  tmd.fitModel(quant_method.getTransformationModel(), quant_method.getTransformationModelParams());
-                  float calculated_feature_ratio = tmd.apply(ratio);
-                  // check for less than zero
-                  if (calculated_feature_ratio < 0.0)
-                  {
-                    calculated_feature_ratio = 0.0;
-                  }
-                  y_fit_data.push_back(calculated_feature_ratio);
-                  result.conc_min = std::min(ratio, result.conc_min);
-                  result.conc_max = std::max(ratio, result.conc_max);
-                  result.feature_min = std::min(calculated_feature_ratio, result.feature_min);
-                  result.feature_max = std::max(calculated_feature_ratio, result.feature_max);
-                }
-                n_points += y_fit_data.size();
-                if (n_points < max_nb_points) {
-                  result.conc_fit_data.push_back(stand_concs_map.at(quant_method.getComponentName()).first);
-                  result.feature_fit_data.push_back(y_fit_data);
-                }
-                else
+        }
+        // Make the line of best fit using the `QuantitationMethods`
+        for (const auto& quant_method : sequence_segment.getQuantitationMethods())
+        {
+          // Skip components that have not been fitted with a calibration curve
+          if (sequence_segment.getComponentsToConcentrations().count(quant_method.getComponentName()) > 0 &&
+              component_names.count(quant_method.getComponentName()) > 0)
+          { 
+            // Make the line of best fit using the `QuantitationMethods`
+            std::vector<float> y_fit_data;
+            if (quant_method.getNPoints() > 0)
+            {
+              for (const auto& ratio : stand_concs_map.at(quant_method.getComponentName()).first) {
+                // TODO: encapsulate in its own method e.g. sequenceSegmentProcessor
+                // TODO: check that the calibration actually found a best fit (and set to all 0 if not)
+                // calculate the absolute concentration
+                OpenMS::TransformationModel::DataPoints data;
+                OpenMS::TransformationDescription tmd(data);
+                tmd.fitModel(quant_method.getTransformationModel(), quant_method.getTransformationModelParams());
```

Comment:
Note that this would be a part of the method I refer to above in the context of the new `CalculateCalibration` method calculates which calculates the line of best fit using the user-specified points.